### PR TITLE
EE: "Internal Error evaluating expression" when instruction does not map to IL offset

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @KevinH-MS, @pnelsonmsft, @ManishJayaswal

**Comment by KevinH-MS** [comment]
When does this happen (what's the end-to-end scenario where an instruction address can be "outside of IL")?

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs [hunk 1]
```diff
@@ -158,6 +160,25 @@ internal sealed class EvaluationContext : EvaluationContextBase
             Guid moduleVersionId,
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs]: Consider:  Can we delete this overload and update the callers to pass "previous == null" to the overload above instead?  It's not a big deal, but it seems like a bit of a gotcha that we have to do the ```NormalizeILOffset``` check in each factory method.

Comment (cston) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs]: We could merge the two overloads. The one difference is that this second overload is used for compilations that contain referenced modules only (see caller). It is unfortunate to require each overload to call NormalizeILOffset but the compiler will report cast errors between uint and int if that's not done. Will leave as two overloads for now.

**Comment by cston** [comment]
@dotnet-bot retest this please

**Comment by cston** [comment]
Customer scenario: When stopped in a managed frame at a native instruction that does not map to an IL instruction, evaluation of expressions including arguments will fail with "Internal Error evaluating expression".

**Comment by MattGertz** [comment]
Approved 6/10 by ML Shiproom.

**Comment by KevinH-MS** [comment]
:+1:

**Pull request merged**